File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP5_projet_1_Sofiane/Projet_1_MEDDOURI_VANSCOOR_premier/Parking.java
public class Parking {
	private static int id = 0;
	private int numParking;
	private Place[] places;
	
	public Parking(Place[] places) {
		this.numParking = id;
		Parking.id ++;
		this.places = places;
	}
	
	public Parking(Parking park) {
		this.numParking = id;
		Parking.id ++;
		this.places = new Place[park.places.length];
		for (int i=0; i<park.places.length; i++) {
			this.places[i] = new Place(park.places[i].getRangee(), park.places[i].getNumPlace());
		}
	}

	public int getNumParking() {
		return numParking;
	}

	public Place[] getPlaces() {
		return places;
	}

	public void setPlaces(Place[] places) {
		this.places = places;
	}
	
	public void addPlace(Place place) {
		Place[] pls = new Place[this.places.length +1];
		for(int i=0; i<this.places.length; i++) {
			pls[i] = new Place(this.places[i]);
		}
		pls[pls.length -1] = new Place(place);
		this.places = pls;
	}
	
	public String placesString() {
		String stringListe = "";
			for(int i=0; i<this.places.length; i++) {
				stringListe += this.places[i].toString() + "\n";
			}
		return stringListe;
	}

	@Override
	public String toString() {
		return "Parking [numParking=" + numParking + ", places=" + this.placesString() + "]";
	}
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Parking.php
<?php
class Parking{

    // attributs

    private static $id = 0;
    private $numeroParking;
    private $places = array();

    // constructeur par initialisation

    public function __construct($places){
        $this->numeroParking = self::$id;
        Parking::$id++;
        $this->places = $places;
    }

    // getters

    public function getNumeroParking(){
        return $this->numeroParking;
    }

    public function getPlace(){
        return $this->places;
    }

    // setter

    public function setNumeroParking(int $numeroParking){
        $this->numeroParking = $numeroParking;
    }

    public function setPlace(Place $place){
        $this->place = $place;
    }

    public function addPlace($place){
        $pls = array();
        $pls = new Place(count($places) + 1);
        for($i = 0; $i < $this->places->length;$i++){
            $pls[i] = new Place($this->places[i]);
        }
        $pls[$pls->length - 1] = new Place($place);
        $this->places = $pls;
    }
/*
    public function placesString(){
        $stringListe = is_numeric("");
        for($i = 0; $i <count($this->places) ;$i++){
            $stringListe += is_numeric($this->places[$i]->__toString()) + is_numeric("\n");
        }
        return $stringListe;
    }
    */

    // methode toString

    public function __toString(){
        return "Parking [numParking=" . $this->numeroParking . "]";
    }
}